pkg/aws/cloudfront/cloudfront.go

```diff
@@ -0,0 +1,140 @@
+package cloudfront
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	cf "github.com/aws/aws-sdk-go/service/cloudfront"
+)
+
+func NewClient(opts ...Option) (*Client, error) {
+	client := &Client{
+		awsCfg:  &aws.Config{},
+		timeout: 30 * time.Second,
+	}
+
+	for _, opt := range opts {
+		opt(client)
+	}
+
+	// By default, if no StaticCredentials are provided, NewSession will use environment variables
+	// AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY, AWS_SESSION_TOKEN
+	session, err := session.NewSession(client.awsCfg)
+	if err != nil {
+		return nil, err
+	}
+
+	client.cfApi = cf.New(session)
+
+	return client, nil
+}
+
+// Returns the current UTC time formatted using the format string
+//	"20060102150405"
+// which means: "2006-01-02 15:04:05"
+func currentTimeSeconds() string {
+	t := time.Now().UTC()
+	return t.Format("20060102150405")
+}
+
+// Verifies the Invalidation struct's pointers are non-nil
+func checkInvalidationStruct(invalidation *cf.Invalidation) error {
```

Comment:
Is the intention that these errors are returned back to the caller?

If so, the caller won't know what to do with these error messages, since they do not set any of these fields.

I think we should re-map these errors to something that can be actionable for a caller of this package.  Maybe these errors are too granular and we only need a subset.

Is it possible for these fields to be entirely empty if the `.CreateInvalidationWithContext` does not return an error?

Reply:
Since AWS api's [Invalidation](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudfront@v1.15.0/types#Invalidation) struct fields are all pointers, my thinking is to have this function as an universal check to make sure there isn't a nil pointer, and subsequent code can safely access without worrying about panic.

If there is no error, I think none of the fields should be set to nil, since they're all labelled as "Required" in the [Invalidation](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudfront@v1.15.0/types#Invalidation) struct.

I made it granular to point out exactly which field is empty, but I see what you mean that it makes it difficult to handle.

I will change this to simply be 2 errors:
- `invalidation` itself is nil
- A field in `invalidation` is nil